hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsBlobClient.java

```diff
@@ -244,8 +314,31 @@ public AbfsRestOperation createPath(final String path,
       final ContextEncryptionAdapter contextEncryptionAdapter,
       final TracingContext tracingContext) throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
+    if (isFile) {
+      AbfsHttpOperation op1Result = null;
+      try {
+        op1Result = getPathStatus(path, false, tracingContext,
```

Comment:
Do we need to have an implicit check happening here?
If not better to call `getPathStatus(path, false, tracingContext, null, false)` This will avoid a list call.

If we need to know if implicit path exists then we can pass true.

I would suggest to check this every where we are doing an internal getPathStatus() and pass the last parameter as needed.

Reply:
Updated